NightscoutServiceKit/NightscoutService.swift

```diff
@@ -375,17 +368,149 @@ extension NightscoutService: RemoteDataService {
         })
     }
     
-    enum NotificationValidationError: LocalizedError {
-        case missingOTP
+    public func handleRemoteNotification(_ notification: [String: AnyObject]) async throws {
+        let commandSource = try commandSource(notification: notification)
+        await commandSource.handleRemoteNotification(notification)
+    }
+    
+    public func processPendingRemoteCommands() async throws {
+        try await commandSourceV2.processPendingRemoteCommands()
+    }
+    
+    private func commandSource(notification: [String: AnyObject]) throws -> RemoteCommandSource {
         
-        var errorDescription: String? {
-            switch self {
-            case .missingOTP:
-                return "Error: Password is required."
-            }
+        guard let versionString = notification["version"] as? String else {
+            return commandSourceV1 //fallback before versions were introduced
+        }
+        
+        guard let version = Double(versionString) else {
+            throw NightscoutServiceError.missingCommandSource
+        }
+        
+        switch version {
+        case 1.0..<2.0:
+            return commandSourceV1
+        case 2.0..<3.0:
+            return commandSourceV2
+        default:
+            throw NightscoutServiceError.missingCommandSource
+        }
+    }
+
+}
+
```

Comment:
A delegate for the command source V1 (pre Remote 2.0 source). It really just handles incoming push notifications.